pkg/server/chaosd/recover.go

```diff
@@ -61,6 +61,7 @@ func (s *Server) RecoverAttack(uid string) error {
 	}
 
 	env := s.newEnvironment(uid)
+	// TODO: In case a scheduled cron is running right now, recovery may lead system to an unknown state
```

Comment:
What about a `pending` state indicating the cron is running now? Or simply give it a lock.

Reply:
So experiment will have `Scheduled` state once created. But on each job run, it'll switch experiment's state to `Pending` and back to `Scheduled` once done?

Reply:
Lock would've been generally a good idea but here recover picks up exp data from DB. I could save running jobs in a map, but not sure how good of an idea that is. :thinking: 
I am thinking to instead check for the last run here and if it's in `RunStarted` state, then don't recover and throw an error. But even with this, it's definitely possible that both recovery and execution happens due to multithreaded server.

Reply:
A similar approach (use a map to store jobs and locks) is adopted in `chaos-mesh/pkg/bpm` https://github.com/chaos-mesh/chaos-mesh/blob/8ad8a6ff66abbceb0707e89e2169b08e8b7692fa/pkg/bpm/bpm.go#L83. I cannot say it's good or bad but you can take a look at it.

Reply:
will check this implementation

Reply:
redundant now since recovery for runs happens after given duration